Clarify not arming PTO for ApplicationData before completion (#3486)
quicwg/base-drafts: draft-ietf-quic-recovery.md

the timer MUST be set for the packet number space with the earliest timeout,
except for ApplicationData, which MUST be ignored until the handshake
completes; see Section 4.1.1 of {{QUIC-TLS}}.  Not arming the PTO for

> ApplicationData prioritizes completing the handshake and prevents the client

Reviewer: Just to clarify a bit more and clean up the run-on sentence.

ApplicationData prioritizes completing the handshake, prevents the client
from sending a 0-RTT packet on a PTO before it knows the server has accepted
0-RTT, and prevents the server from sending a 1-RTT packet on a PTO before
before it has the keys to process a 1-RTT acknowledgment.

Author: Accepted

from sending a 0-RTT packet on a PTO before it knows the server has accepted
0-RTT, and prevents the server from sending a 1-RTT packet on a PTO before

> it has the keys to process a 1-RTT acknowledgment.

Reviewer 2: ```suggestion
ApplicationData prioritizes completing the handshake, prevents a client
from retransmitting a 0-RTT encrypted packet on a PTO expiration before
confirming that the server is able to decrypt 0-RTT packets, and prevents
a server from sending a 1-RTT-encrypted packet on a PTO expiration before
it has the keys to process an acknowledgement received in a 1-RTT-enrypted
packet.
```

Reviewer 3: That is a very long sentence.

> it has the keys to process a 1-RTT acknowledgment.

Reviewer 3: ```suggestion
ApplicationData prioritizes completing the handshake, prevents a client from
retransmitting a 0-RTT packet on a PTO expiration before confirming that the
server is able to decrypt 0-RTT packets, and prevents a server from sending a
1-RTT packet on a PTO expiration before it has the keys to process an
acknowledgement.
```

> has the keys to process an acknowledgement.

Reviewer 2: ```suggestion
and it also prevents a server from sending a 1-RTT packet on a PTO expiration
before it has the keys to process an acknowledgement.
```

> has the keys to process an acknowledgement.

Reviewer 2: ```suggestion
ApplicationData prevents a client from retransmitting a 0-RTT packet in response
before confirming that the server is able to decrypt 0-RTT packets, and it prevents
a server from sending a 1-RTT packet in response before it has the keys to process
an acknowledgement.
```

> has the keys to process an acknowledgement.

Reviewer 3: Not arming the PTO for
```suggestion
ApplicationData prevents a client from sending a packet on PTO where there is uncertainty about the ability of the server to acknowledge the packet.
```

> has the keys to process an acknowledgement.

Reviewer 2: ```suggestion
ApplicationData prevents a client from sending a packet on PTO where there is uncertainty about the ability of the server to acknowledge the packet. This can happen before a client confirms that the server is able to decrypt 0-RTT packets, and also at a server before it has the keys to process an acknowledgement.
```

Author: This isn't only about the server acknowledging packets, so I think this isn't quite correct.

> before it has the keys to process a 1-RTT acknowledgment.

Reviewer 4: redundant before?

Author: Fixed, thanks!
